pkg/core/processor/httpproc.go

```diff
@@ -0,0 +1,134 @@
+package processor
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"log/slog"
+
+	"github.com/ksysoev/deriv-api-bff/pkg/core"
+	"github.com/ksysoev/deriv-api-bff/pkg/core/request"
+)
+
+type HTTPProc struct {
+	name        string
+	method      string
+	urlTemplate *template.Template
+	tmpl        *template.Template
+	fieldMap    map[string]string
+	allow       []string
+}
+
+// NewHTTP creates a new instance of HTTPProc based on the provided configuration.
+// It takes a single parameter cfg of type *Config which contains the necessary configuration details.
+// It returns a pointer to an HTTPProc initialized with the values from the configuration.
+func NewHTTP(cfg *Config) *HTTPProc {
```

Comment:
same - maybe return type can be `*Processor`

Reply:
will help in mocking - since Processor is interface but NewHTTP is struct

Reply:
same here, we should try to return concrete types and let caller to decide if the want to apply any interface to it to limit abstraction.